include/bluetooth/mesh/dk_prov.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+/**
+ * @file
+ * @brief Bluetooth Mesh Provisioning handler for Nordic DKs.
+ * @defgroup bt_mesh_dk_prov Bluetooth Mesh Provisioning handler for Nordic DKs
+ * @{
+ */
+
+#ifndef BT_MESH_DK_PROV_H__
+#define BT_MESH_DK_PROV_H__
+
+#include <bluetooth/mesh.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/** @brief Initialize the provisioning handler.
+ *
+ * @return The provisioning properties to pass to @ref bt_mesh_init.
+ */
+const struct bt_mesh_prov *bt_mesh_dk_prov_init(void);
```

Comment:
Would it make sense to have the API independant of the DK?
So we could have dk_prov.c, nfc_prov.c, both implementing prov.h, and from Kconfig we could select which one is added to the build?

Reply:
Good question.

The goal of this module is to provide a simple implementation of provisioning for the DKs to avoid duplicating it in every example. I don't really want people to use this for other things than board bring up and demos. In a real deployment, they'll have custom hardware and should create their own handling, and most likely, they'll need to change things in the API as well.

I'd rather keep this module as simple as possible, and instruct users to implement their own handlers when they get out of the prototype phase.

If we add nfc_prov, it should definitely be a reusable general library though.